Packet++/header/TcpLayer.h

```diff
@@ -306,44 +392,95 @@ namespace pcpp
 		 * @param[in] optionValue A buffer containing the option value. This buffer is read-only and isn't modified in any way.
 		 * @param[in] optionValueLen Option value length in bytes
 		 */
+		PCPP_DEPRECATED_MSG("enum TcpOptionType is deprecated; Use enum class TcpOptionKind instead")
 		TcpOptionBuilder(TcpOptionType optionType, const uint8_t* optionValue, uint8_t optionValueLen) :
 			TLVRecordBuilder((uint8_t)optionType, optionValue, optionValueLen) {}
 
+		/**
+		 * A c'tor for building TCP options which their value is a byte array. The TcpOption object can be later
+		 * retrieved by calling build()
+		 * @param[in] optionType TCP option type
+		 * @param[in] optionValue A buffer containing the option value. This buffer is read-only and isn't modified in any way.
+		 * @param[in] optionValueLen Option value length in bytes
+		 */
+		TcpOptionBuilder(TcpOptionKind optionType, const uint8_t* optionValue, uint8_t optionValueLen) :
+			TLVRecordBuilder(static_cast<uint8_t>(optionType), optionValue, optionValueLen) {}
+
 		/**
 		 * A c'tor for building TCP options which have a 1-byte value. The TcpOption object can be later retrieved
 		 * by calling build()
 		 * @param[in] optionType TCP option type
 		 * @param[in] optionValue A 1-byte option value
 		 */
+		PCPP_DEPRECATED_MSG("enum TcpOptionType is deprecated; Use enum class TcpOptionKind instead")
```

Comment:
you repeat this `PCPP_DEPRECATED_MSG("enum TcpOptionType is deprecated; Use enum class TcpOptionKind instead")` many times, maybe consider making it as a macro?

e.g. `PCPP_DEPRECATED_TCP_OPTION_TYPE TcpOptionBuilder()`